File: frontend/src/types/parser.ts
export interface ParserInfo {
  parser_id: string;
  name: string;
  description: string;
  supported_extensions: string[];
}

export interface PipelineStepInfo {
  step_id: string;
  name: string;
  description: string;
  applicable: string[];
  default_order: number;
}

export interface PipelineStepsResponse {
  file_kind: string;
  preprocess: PipelineStepInfo[];
  postprocess: PipelineStepInfo[];
  presets: {
    tutorial_full: string[];
    none: string[];
  };
}

export interface LogItem {
  level: string;
  message: string;
  timestamp?: string | null;
}

export interface ErrorItem {
  code: string;
  message: string;
  detail?: string | null;
}

export interface PageResult {
  page_no: number;
  text: string;
  blocks: Record<string, unknown>[];
}

export interface TableResult {
  table_id: string;
  page_no: number;
  rows: string[][];
}

export interface ParseResponse {
  success: boolean;
  parser_id: string;
  file_name: string;
  extension: string;
  elapsed_ms: number;
  page_count?: number | null;
  text_length: number;
  table_count: number;
  error_count: number;
  text: string;
  pages: PageResult[];
  tables: TableResult[];
  logs: LogItem[];
  errors: ErrorItem[];
}

export interface SelectedFileInfo {
  file: File;
  name: string;
  extension: string;
  size: number;
  mimeType: string;
}

export type ResultTab =
  | "text"
  | "json"
  | "log"
  | "error"
  | "table"
  | "compare";

export type RunStatus = "idle" | "ready" | "running" | "success" | "failed";

File: backend/app/schemas/parser.py
from typing import Any

from pydantic import BaseModel, Field


class ParserInfo(BaseModel):
    parser_id: str
    name: str
    description: str
    supported_extensions: list[str]


class LogItem(BaseModel):
    level: str
    message: str
    timestamp: str | None = None


class ErrorItem(BaseModel):
    code: str
    message: str
    detail: str | None = None


class PageResult(BaseModel):
    page_no: int
    text: str
    blocks: list[dict[str, Any]] = Field(default_factory=list)


class TableResult(BaseModel):
    table_id: str
    page_no: int
    rows: list[list[str]]


class ParseResponse(BaseModel):
    success: bool
    parser_id: str
    file_name: str
    extension: str
    elapsed_ms: int
    page_count: int | None = None
    text_length: int = 0
    table_count: int = 0
    error_count: int = 0
    text: str = ""
    pages: list[PageResult] = Field(default_factory=list)
    tables: list[TableResult] = Field(default_factory=list)
    logs: list[LogItem] = Field(default_factory=list)
    errors: list[ErrorItem] = Field(default_factory=list)
